frontend/src/metabase-lib/filter.ts

```diff
@@ -100,6 +102,29 @@ export function removeFilters(query: Query, stageIndex: number): Query {
   );
 }
 
+/**
+ * This function should live in metabase/querying/utils/filters/utils.ts
+ * but needs to be used in metabase-lib/v1 which would cause a forbidden import.
+ * TODO: Move it there when metabase-lib/v1 is no more.
+ */
+export function appendStageIfSummarized(query: Query): Query {
+  const isSummarized =
+    aggregations(query, -1).length > 0 && breakouts(query, -1).length > 0;
```

Comment:
This looks confusing. Judging by the name `appendStageIfSummarized`, I would expect aggregation _or_ breakouts, not necessarily both.

Reply:
> This looks confusing. Judging by the name `appendStageIfSummarized`, I would expect aggregation _or_ breakouts, not necessarily both. I'm not sure what a good concise name would be, unfortunately.

Agreed.
@metamben How about `appendFilteringStageIfNecessary`?
This name indicates the purpose of the stage and hides the implementation details.

> But if we only have breakouts, then filtering the rows could in theory make sense.

True. I wonder how many users have questions with stages that have breakouts but no aggregations.
Anyway, this is how it works in the Filter Modal today, and the goal of this project is to expose the same columns for filtering in Dashboards as we do in the Filter Modal. Changing this would need confirmation from the product team.

Reply:
`appendFilteringStageIfNecessary` is much better. Would `ensureFilteringStage` make sense? That's more concise. Whichever name we go with, the notion of "filtering stage" should be documented, I guess.

Reply:
I see that you went with `ensure-filter-stage` - sounds good :+1: 


af72b8d805c2485c8a5b4de155687526bc52ef55

Reply:
I'll use the ML function in a separate PR.
Merging this PR to the integration branch right now.